### PR TITLE
Data: Add a select interface with collections for IcebergGenerics

### LINKED ISSUE #6833: [Java Open Api] Add select override method in IcebergGenerics
### Feature Request / Improvement

The IcebergGenerics class only has String ... method for **select**, should add Collection<String> for **select** will be better!

### Query engine

Other

### PR DESCRIPTION
**Pull request by hunter-cloud09**
The IcebergGenerics class only has String ... method for select, should add Collection for select will be better!

closes #6833

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Review by Fokko — APPROVED** [review]
Makes sense to me

**Comment by hunter-cloud09** [comment]
@nastra @Fokko thx for your review, This problem is only encountered when I use it. I will check the features of TableScan later, and then add it to IcebergGenerics.

File: data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java [hunk 1]
```diff
@@ -66,6 +67,11 @@ public ScanBuilder select(String... selectedColumns) {
       return this;
     }
 
+    public ScanBuilder select(Collection<String> columns) {
+      this.tableScan = tableScan.select(ImmutableList.copyOf(columns));
```
Comment (ajantha-bhat) [inline data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java:71]: Just wondering do we really need a copy of the collection again? Can't we pass the collection directly as the below interface supports it?
https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/Scan.java#L87

Comment (hunter-cloud09) [reply data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java:71]: In my opinion, there is no need to copy, but I am not sure why the previous author did this? Is there any special purpose?

Comment (ajantha-bhat) [reply data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java:71]: they may want to convert arrays into a collection and use the collection interface. But that is also not needed now as the underlying interface has a conversion logic. 

https://github.com/apache/iceberg/blob/634215fcd411dd8a6d5c0b6d7d8dbf4eb66494d8/api/src/main/java/org/apache/iceberg/Scan.java#L96-L98

Comment (hunter-cloud09) [reply data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java:71]: get it, i'll change there

**Review by ajantha-bhat — APPROVED** [review]
LGTM. 

PR title can be "Data: Add a select interface with collections for IcebergGenerics"

**Comment by hunter-cloud09** [comment]
> LGTM.
> 
> PR title can be "Data: Add a select interface with collections for IcebergGenerics"

Get It!

**Comment by hunter-cloud09** [comment]
@ajantha-bhat Is there anything else I need to do before merging?

**Comment by Fokko** [comment]
Thanks @hunter-cloud09 for adding this, and @ajantha-bhat @nastra for the review! 👏🏻

**Pull request merged**